Calibration/EcalCalibAlgos/src/ECALpedestalPCLworker.cc

```diff
@@ -0,0 +1,121 @@
+    digiTokenEE_ = consumes<EEDigiCollection>(digiTagEE_);
+}
+
+
+
+// ------------ method called for each event  ------------
+void
+ECALpedestalPCLworker::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+    using namespace edm;
+      
+    Handle<EBDigiCollection> pDigiEB;
+    iEvent.getByLabel("ecalDigis","ebDigis",pDigiEB);
+   
+    Handle<EEDigiCollection> pDigiEE;
+    iEvent.getByLabel("ecalDigis","eeDigis",pDigiEE);
+
+
+
+    for (EBDigiCollection::const_iterator pDigi=pDigiEB->begin(); pDigi!=pDigiEB->end(); ++pDigi){
+         
+        EBDetId id = pDigi->id();
+        uint32_t hashedId = id.hashedIndex();
+
+        EBDataFrame digi( *pDigi );
+
+        uint16_t maxdiff = *std::max_element(digi.frame().begin(), digi.frame().end(), adc_compare )  - 
+                           *std::min_element(digi.frame().begin(), digi.frame().end(), adc_compare ); 
+        if ( maxdiff> kThreshold ) continue; // assume there is signal in this frame       
+        //for (auto& mgpasample : digi.frame()) meEB_[hashedId]->Fill(mgpasample&0xFFF);
+        for (edm::DataFrame::iterator mgpasample = digi.frame().begin();  
+             mgpasample!=digi.frame().begin()+kPedestalSamples; 
+             ++mgpasample )
+            meEB_[hashedId]->Fill(*mgpasample&0xFFF);
+
+    } // eb digis
+   
+
+    for (EEDigiCollection::const_iterator pDigi=pDigiEE->begin(); pDigi!=pDigiEE->end(); ++pDigi){
+         
+        EEDetId id = pDigi->id();
+        uint32_t hashedId = id.hashedIndex();
+
+        EBDataFrame digi( *pDigi );
+
+        uint16_t maxdiff = *std::max_element(digi.frame().begin(), digi.frame().end(), adc_compare )  - 
+                           *std::min_element(digi.frame().begin(), digi.frame().end(), adc_compare ); 
+        if ( maxdiff> kThreshold ) continue; // assume there is signal in this frame       
+        //for (auto& mgpasample : digi.frame()) meEE_[hashedId]->Fill(mgpasample&0xFFF);
+        for (edm::DataFrame::iterator mgpasample = digi.frame().begin();  
+             mgpasample!=digi.frame().begin()+kPedestalSamples; 
+             ++mgpasample )
+            meEE_[hashedId]->Fill(*mgpasample&0xFFF);
+
+    } // ee digis
+
+
+
+   
+}
+
+
+void 
+ECALpedestalPCLworker::beginJob()
+{
+}
+
+
+void 
+ECALpedestalPCLworker::endJob() 
+{
+}
+
+
+void
+ECALpedestalPCLworker::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.setUnknown();
+    descriptions.addDefault(desc);
+}
+
+
+void 
+ECALpedestalPCLworker::bookHistograms(DQMStore::IBooker & ibooker, edm::Run const & run, edm::EventSetup const & es){
+    
+    ibooker.cd();
+    ibooker.setCurrentFolder("EcalCalibration/EcalPedestalPCL");
+
+    for ( uint32_t i = 0 ; i< EBDetId::kSizeForDenseIndexing; ++i){
+        std::stringstream hname;
+        hname<<"eb_"<<i;       
+        meEB_.push_back(ibooker.book1D(hname.str(),hname.str(),100,150,250));
```

Comment:
most channels do have pedestal centered in the neighborhood of 200
this 1d histo booking will catch all of those se
have you considered centering the histograms dynamically, based on the current value of the pedestal ?